keras/engine/training.py

```diff
@@ -964,9 +978,38 @@ def _make_predict_function(self):
                                                name='predict_function',
                                                **kwargs)
 
```

Comment:
Since these functions below are platform specific, perhaps they should be implemented in the backend instead?